### PR TITLE
ByteArrayCounter, 	feedback tracking commander

### PR DESCRIPTION
**Pull request by yingjieMiao**
@stkem 

this deals with storage/update. 
Inference/boosting part comes later.

### CODE REVIEW

File: kifi-backend/modules/curator/app/com/keepit/curator/feedback/ByteArrayCounter.scala [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    val arr = toArray()
+    (indexes zip values).foreach { case (i, x) => arr(i) = x }
+    ByteArrayCounter.fromArray(arr)
+  }
+
+  def toArray(): Array[Int] = Array.tabulate(bytes.size) { i => get(i) }
+}
+
+object ByteArrayCounter {
+  val MAX_COUNT = 255
```
Comment (stkem) [inline kifi-backend/modules/curator/app/com/keepit/curator/feedback/ByteArrayCounter.scala]: Why is this configurable? Won't it always be 255? (being a byte, after all)

File: kifi-backend/modules/curator/app/com/keepit/curator/feedback/ByteArrayCounter.scala [hunk 2]
```diff
@@ -0,0 +1,73 @@
+package com.keepit.curator.feedback
+
+import play.api.libs.json._
+
+// Immutable operations. Optimized for thread-safe.
+case class ByteArrayCounter(bytes: Array[Byte]) {
```
Comment (stkem) [inline kifi-backend/modules/curator/app/com/keepit/curator/feedback/ByteArrayCounter.scala]: Why a `case class` instead of a regular `class`?

**Comment by stkem** [comment]
lgtm overall, but how will we know if rescaling happened? Seems important.

File: kifi-backend/modules/curator/app/com/keepit/curator/feedback/ByteArrayCounter.scala (hunk 1 above)
Comment (yingjieMiao) [inline kifi-backend/modules/curator/app/com/keepit/curator/feedback/ByteArrayCounter.scala]: more like a syntatic sugar for the constant 255. Named constants are easier to search, and prevents typo. As you can see, the constant is used in a few places.

File: kifi-backend/modules/curator/app/com/keepit/curator/feedback/ByteArrayCounter.scala (hunk 2 above)
Comment (yingjieMiao) [inline kifi-backend/modules/curator/app/com/keepit/curator/feedback/ByteArrayCounter.scala]: You are right. I will change it to regular class to restrict access to those bytes.

**Comment by stkem** [comment]
:+1:

**Pull request merged**